src/Tools/IdeBenchmarks/PersistentStorageBenchmarks.cs

```diff
@@ -0,0 +1,135 @@
+        private ComposableCatalog _catalog;
+        private IExportProviderFactory _exportProviderFactory;
+
+        [Params(0, 1000, 10000)]
+        public int ClassCount { get; set; }
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            _catalog = TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic.WithPart(typeof(PersistentStorageLocationService));
+            _exportProviderFactory = ExportProviderCache.GetOrCreateExportProviderFactory(_catalog);
+        }
+
+        [IterationSetup]
+        public void IterationSetup()
+            => _useExportProviderAttribute.Before(null);
+
+        [IterationCleanup]
+        public void IterationCleanup()
+            => _useExportProviderAttribute.After(null);
+
+        [Benchmark]
+        public object TestProject()
+        {
+            var files = CreateTestInput(fileCount: ClassCount);
+            return WriteStorageAsync(files, parseOptions: null).Result;
+        }
+
+        private static XElement CreateTestInput(int fileCount)
+        {
+            var documents = new XElement[fileCount];
+            for (var i = 0; i < fileCount; i++)
+            {
+                documents[i] = new XElement("Document",
+                    new XAttribute("FilePath", $"Class{i}.cs"),
+                    new XText($@"
+class Class{i}
+{{
+}}
+"));
+            }
+
+            return new XElement(
+                "Workspace",
+                new XAttribute("FilePath", "SolutionPath.sln"),
+                new XElement(
+                    "Project",
+                    new XAttribute("AssemblyName", "CSharpAssembly"),
+                    new XAttribute("Language", LanguageNames.CSharp),
+                    new XAttribute("CommonReferences", "true"),
+                    documents));
+        }
+
+        private async Task<SymbolTreeInfo> WriteStorageAsync(XElement workspaceElement, ParseOptions parseOptions)
+        {
+            using (var workspace = TestWorkspace.Create(workspaceElement.ToString(), exportProvider: _exportProviderFactory.CreateExportProvider()))
+            {
+                workspace.Options = workspace.Options.WithChangedOption(StorageOptions.SolutionSizeThreshold, -1);
+
+                var persistentStorageService = workspace.Services.GetRequiredService<IPersistentStorageService>();
+                var persistentStorage = persistentStorageService.GetStorage(workspace.CurrentSolution);
+                if (persistentStorage is NoOpPersistentStorage)
+                {
+                    throw new InvalidOperationException("Benchmark is not configured to use persistent storage.");
+                }
+
+                var symbolTreeInfoCacheService = workspace.Services.GetRequiredService<ISymbolTreeInfoCacheService>();
+                var symbolTreeInfo = await symbolTreeInfoCacheService.TryGetSourceSymbolTreeInfoAsync(workspace.CurrentSolution.Projects.Single(), CancellationToken.None);
```

Comment:
@heejaechang This is not working because the incremental analyzer is not forced to run before this point. How can I force the incremental analyzer to run?

Reply:
you can use this 

http://index/?leftProject=Microsoft.CodeAnalysis.Features&leftSymbol=ulsbleozj4aj&file=SolutionCrawler%5CSolutionCrawlerRegistrationService.cs&line=145

if you have access to internal.

Reply:
if you just want to wait until everything done, then, you can use async listener to wait until solution crawler says it is all done.

basically, wait for workspace and solution crawler to be done.
get IAsynchronousOperationListenerProvider and do provider.GetAllAsync(new string[] { FeatureAttributes.Workspace, FeatureAttributes.SolutionCrawler)

Reply:
@heejaechang benchmarks run thousands of times and expect precise timing for the actual operations; waits are detrimental to the measurements.

Reply:
➡️ Moving persistent storage benchmarks to a future change